### DISCUSSION TITLE
How to get the latest coordinates of control points？

### DISCUSSION THREAD

**Discussion by wgtfly** [mdolab/dafoam#719] [Tutorials]
Hi professor,could you tell me where I can see the latest control point coordinates? Or where they are stored?

**Comment by friedenhe** [comment]
The control point(FFD) displacements should be printed to the screen during optimization

**Reply by wgtfly** [reply]
So the coordinates are not saved, and the shape of the intermediate process does not seem to be saved either. Is that right? So if I want to see the entire shape change process, do I need an additional script to achieve this?

**Reply by benoitpaillard** [reply]
this is supposed to be obtained by writeDeformedFFDs, which is broken according to [this](https://github.com/mdolab/dafoam/discussions/456)

A workaround is to use `writeTecplot` or `writePlot3d`, for instance in NACA0012 tutorial :

```
prob.model.geometry.DVGeo.writeTecplot("deformedFFD.dat")
prob.model.geometry.DVGeo.writePlot3d("deformedFFD.xyz")
```

**Reply by wgtfly** [reply]
I see, thank you for sharing this, I will try.
